node/src/cli.rs

```diff
@@ -16,9 +16,70 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-use sc_cli::{KeySubcommand, RunCmd, SignCmd, VanityCmd, VerifyCmd};
+use sc_cli::{KeySubcommand, SignCmd, VanityCmd, VerifyCmd};
+use std::{fmt, str::FromStr};
 use structopt::StructOpt;
 
+#[derive(Debug, Clone)]
+pub struct RuntimeInstanceError(String);
+
+impl fmt::Display for RuntimeInstanceError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let RuntimeInstanceError(message) = self;
+        write!(f, "RuntimeInstanceError: {}", message)
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub enum RuntimeInstance {
+    Main,
+    Staking,
+}
+
+impl RuntimeInstance {
+    fn variants() -> [&'static str; 2] {
+        ["main", "staking"]
+    }
+
+    pub fn is_staking_runtime(&self) -> bool {
+        match self {
+            Self::Main => false,
+            Self::Staking => true,
+        }
+    }
+}
+
+impl fmt::Display for RuntimeInstance {
```

Comment:
This trait should be accessible via `#[derive(fmt::Display)]` unless the manual implementation is preferred.

Reply:
As suggested, extra CLI options removed :+1: